BLSQ/iaso
# POLIO-799 override step timewidget front end

Update Front-end i polio budget to display skipped/cancelled steps

Related JIRA tickets : POLIO-799

## Self proof reading checklist


## Changes

- Add specific icon for skipped/cancelled steps
- Add tooltip on icon with explanation of the skipped/cancelled step
- Add some TS typing
- Feature:
   - When moving forward: 
       - undone mandatory steps are shown as skipped
       - done cancelled steps remain cancelled
   - When moving backwards:
       - undone mandatory steps are greyed out, but not cancelled (as they still need to be performed)
       - done mandatory steps are marked as cancelled 

## How to test
- use the config from [the drive](https://drive.google.com/drive/u/0/folders/1ua4yzyFOGayweQVNKP0hyBNDKhQrPlx1)
- create a campaign and set it up for budget
- make sure you're set up to create any event on the workflow, and that you are a super user (so you have access to the override step)
- create a few steps in the budget
- override to either go forward or backwards
    - going forward: 
        - Steps should be skipped correctly
        - The progress bar should be completed for all complete sections, including those that have skipped steps
    - going backwards:
        - Cancelled steps should be marked correctly
        - Sections of the progress bar that the cancelled steps made incomplete should appear as incomplete    

## Print screen / video

https://user-images.githubusercontent.com/38907762/216964128-ef32addd-a339-401a-9b32-4a6e2b6a01ee.mov


## Notes

merges into https://github.com/BLSQ/iaso/pull/356
